ref/haskell/test/Spec.hs

```diff
@@ -73,6 +86,10 @@ tests = testGroup "Tests"
                 let checksumEncoded = bech32Encode resultHRP resultData
                     expectedChecksum = Just $ BSC.map toLower checksum
                 assertEqual (show checksum ++ " re-encode") expectedChecksum checksumEncoded
+    , testCase "Invalid checksums" $ forM_ invalidChecksums $ \checksum -> do
+          case bech32Decode checksum of
+            Just (_, _) -> assertFailure (show checksum)
```

Comment:
Simpler: `Just _ -> `

Reply:
forM_ invalidChecksums $ 
    \checksum -> assertBool (show checksum) (isNothing $ bech32Decode checksum)

Similar to `Invalid Addresses` below.